app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PiecesController < ApplicationController
+  def update
+    @piece = Pieces.find(params[:id])
+    new_x = params[:x].to_i
+    new_y = params[:y].to_i
+    return update_attributes(x_position: new_x, y_position: new_y) if @piece.move!(new_x, new_y)
```

Comment:
Can you move the `update_attributes` part into the `move!` method itself?  In general, controllers are for "controlling" the request, or to put it another way they're responsible for directing what the request should do.  The controller should be translating the request that the user made into _what_ should happen, but the models should, in general, do the heavy lifting.

So this would just become:

```ruby
return @piece.move!(new_x, new_y)
```